chalice/deploy/packager.py

```diff
@@ -0,0 +1,213 @@
+import hashlib
+import inspect
+import os
+import shutil
+import subprocess
+import sys
+import zipfile
+
+import virtualenv
+from typing import Any, Optional  # noqa
+
+import chalice
+from chalice import compat, app
+
+
+class LambdaDeploymentPackager(object):
+    _CHALICE_LIB_DIR = 'chalicelib'
+    _VENDOR_DIR = 'vendor'
+
+    def _create_virtualenv(self, venv_dir):
+        # type: (str) -> None
+        # The original implementation used Popen(['virtualenv', ...])
+        # However, it's hard to make assumptions about how a users
+        # PATH is set up.  This could result in using old versions
+        # of virtualenv that give confusing error messages.
+        # To fix this issue, we're calling directly into the
+        # virtualenv package.  The main() method doesn't accept
+        # args, so we need to patch out sys.argv with the venv
+        # dir.  The original sys.argv is replaced on exit.
+        original = sys.argv
+        sys.argv = ['', venv_dir, '--quiet']
```

Comment:
Why not use ``virtualenv. create_environment()`` instead of monkeypatching ``sys.argv``?